src/java.base/share/classes/java/lang/doc-files/ValueBased.html

```diff
@@ -31,36 +31,42 @@
         solely from the instance's state and not from its identity or the state
         of any other object or variable;</li>
-    <li>make no use of identity-sensitive operations such as reference
-        equality (<code>==</code>) between instances, identity hash code of
-        instances, or synchronization on an instances's intrinsic lock;</li>
-    <li>are considered equal solely based on <code>equals()</code>, not
-        based on reference equality (<code>==</code>);</li>
-    <li>do not have accessible constructors, but are instead instantiated
-        through factory methods which make no commitment as to the identity
-        of returned instances;</li>
-    <li>are <em>freely substitutable</em> when equal, meaning that interchanging
-        any two instances <code>x</code> and <code>y</code> that are equal
-        according to <code>equals()</code> in any computation or method
-        invocation should produce no visible change in behavior.
+    <li>have instances that are <em>freely substitutable</em> when equal, meaning
+        that interchanging any two instances <code>x</code> and <code>y</code> that
+        are equal according to <code>equals()</code> produces no visible change in
+        the behavior of the class's methods;
     </li>
+    <li>perform no synchronization on an instance's intrinsic lock;</li>
+    <li>do not have (or have deprecated any) accessible constructors;</li>
+    <li>may support instance creation through factory methods that do <em>not</em>
+        promise a unique identity for each invocation—in particular, each factory
+        method must allow for the possibility that if two independently-produced
+        instances are equal according to <code>equals()</code>, they may also be
+        equal according to <code>==</code>;
+    </li>
+    <li>are final;</li>
+    <li>extend either <code>Object</code> or a hierarchy of abstract classes that
+        declare no instance fields or instance initializers and whose constructors
+        are empty.</li>
+    <li>
 </ul>
 
-<p>A program may produce unpredictable results if it attempts to distinguish two
-    references to equal values of a value-based class, whether directly via reference
+<p>When two instances of a value-based class are `equal`, a program should not attempt
+    to distinguish between their identities, whether directly via reference
     equality or indirectly via an appeal to synchronization, identity hashing,
-    serialization, or any other identity-sensitive mechanism.  Use of such
-    identity-sensitive operations on instances of value-based classes may have
-    unpredictable effects and should be avoided.</p>
+    serialization, or any other identity-sensitive mechanism.</p>
+<p>Synchronization on instances of value-based classes is strongly discouraged,
+    because the programmer cannot usually guarantee unique ownership of the
+    associated lock.</p>
```

Comment:
I don't think the phrase: 
`because the programmer cannot usually guarantee unique ownership of the associated lock.`
adds anything and is obscure.  Until the semantics of object change, there is no change in the synchronization behavior.
For Valhalla value-based instances, there is explicitly no (implicitly) associated lock, so synchronization is illegal.

Reply:
The point is that if you can't prove you have a unique identity (per the factory method clause), then you can't guarantee that you have exclusive control over the monitor, so synchronization is dangerous. A variant of the text discouraging synchronization was there before any design discussions about primitive object semantics.

I added "usually" because of the possibility that someone can prove that a field value they provide is unique (according to 'equals') and thus can prove that the value-based class instance is also unique. But when I think about actual examples of value-based classes, that seems quite unlikely, and it would probably be clearer to remove the "usually".